### PR TITLE
Update tool to upload multiple files 

### PR DESCRIPTION
**Pull request by carlosu7**
Rename `file` property to `files` and update endpoint to upload multiple files according to pixee platform changes

### CODE REVIEW

File: dist/index.js
Comment (carlosu7) [inline dist/index.js:1]: just for testing purposes

File: src/action.ts [hunk 1]
```diff
@@ -94,8 +94,9 @@ async function fetchOrLocateResultsFile(tool: Tool, results: any, fileName: stri
 async function fetchSonarCloudIssues(){
   const sonarCloudInputs = getSonarCloudInputs();
   const results = await retrieveSonarCloudIssues(sonarCloudInputs);
+
   core.info(
-    `Found ${results.total} SonarCloud issues for component ${sonarCloudInputs.componentKey}`
+    `HARDCODED Found ${results.total} SonarCloud issues for component ${sonarCloudInputs.componentKey}`
```
Comment (drdavella) [inline src/action.ts:99]: Is this just for debugging?

File: src/action.ts [hunk 2]
```diff
@@ -71,9 +71,9 @@ async function fetchOrLocateContrastResultsFile() {
   throw new Error("Contrast requires a file to be provided");
 }
 
-async function fetchOrLocateSonarResultsFile(resultType : SONAR_RESULT) {
+async function fetchOrLocateSonarResultsFile(resultType : SONAR_RESULT, index ?: number) {
```
Comment (drdavella) [inline src/action.ts:74]: Is this method actually called with this `index` parameter anywhere? I'm not sure I understand the reason for adding it.

File: src/pixee-platform.ts [hunk 3]
```diff
@@ -5,14 +5,20 @@ import FormData from "form-data";
...
-  const fileContent = fs.readFileSync(file, "utf-8");
+export async function uploadInputFile(tool: TOOL_PATH, files: Array<string>) {
   const form = new FormData();
-  form.append("file", fileContent);
-  const pixeeUrl = core.getInput("pixee-api-url");
 
+  // Append each file to the form data
+  files.forEach(file => {
+    const fileContent = fs.readFileSync(file);
+    form.append("files", fileContent, file);
```
Comment (drdavella) [inline src/pixee-platform.ts:14]: I think this is using the entire path and we might want to use just the base filename?

**Review by drdavella — COMMENTED** [review]
Thanks for your patience on this review 😅. I've been holding off since we can't use this until the changes on the platform side have been deployed.

If I'm understanding this change correctly, it looks like we're not actually capturing paginated results for any of the supported tools yet? That's okay but we should work to add support for Sonar issues and hotspots next.

File: src/action.ts (hunk 1 above)
Comment (carlosu7) [reply src/action.ts:99]: i forgot to revert this change

File: src/action.ts (hunk 2 above)
Comment (carlosu7) [reply src/action.ts:74]: no, it was for testing purposes when I was uploading multiple files

File: src/pixee-platform.ts (hunk 3 above)
Comment (carlosu7) [reply src/pixee-platform.ts]: i have addressed this

File: __tests__/action.test.ts [hunk 4]
```diff
@@ -174,10 +174,7 @@ describe("action", () => {
 
       expect(retrieveSonarCloudIssuesMock).toHaveBeenCalled();
       expect(retrieveSonarCloudHotspotsMock).toHaveBeenCalled();
-      expect(uploadInputFileMock).toHaveBeenCalledWith(
-        "sonar_issues",
-        expect.stringMatching(/sonar-issues.json$/)
-      );
+      expect(uploadInputFileMock).toHaveBeenCalled();
```
Comment (drdavella) [inline __tests__/action.test.ts:177]: Is there a reason to make this assertion less specific? Could we assert that it got called with the expected array contents?

File: src/pixee-platform.ts [hunk 5]
```diff
@@ -5,14 +5,23 @@ import FormData from "form-data";
...
   const form = new FormData();
-  form.append("file", fileContent);
-  const pixeeUrl = core.getInput("pixee-api-url");
 
+  const path = require('path');
+
+  // Append each file to the form data
+  files.forEach(file => {
+    const fileContent = fs.readFileSync(file);
+  const baseFilename = path.basename(file);
```
Comment (drdavella) [inline src/pixee-platform.ts:16]: Need to fix indentation here I think, although you probably don't need the intermediate variable at all.

**Comment by drdavella** [comment]
@carlosu7 does it make sense to ship this change without actually implementing pagination for anything? Or should we just go ahead and do this when we implement Sonar pagination?

Is the platform backwards-compatible with the `file` parameter?

**Comment by carlosu7** [comment]
> @carlosu7 does it make sense to ship this change without actually implementing pagination for anything? Or should we just go ahead and do this when we implement Sonar pagination?
> 
> Does the platform change remain backwards-compatible with the `file` parameter?

@drdavella i opened this PR because of the renaming of the form data `file` to `files`, but since your comment about keeping it as `file`, this PR is not required now.

We can hold this PR until pagination is ready, what do you think?

**Comment by drdavella** [comment]
@carlosu7 yes if you want to keep it open and add pagination for Sonar when it's ready, please just request my review again.

**Comment by carlosu7** [comment]
@drdavella it is ready to review, here you can see a test with the following values:
page size = 1
total of sonar issues = 11
expected number of files: 11

https://github.com/carlosu7/WebGoat_12_23/actions/runs/9670042632/job/26677944207

File: src/action.ts [hunk 6]
```diff
@@ -12,6 +12,13 @@ import { getDefectDojoInputs, retrieveDefectDojoResults } from "./defect-dojo";
 import { getGitHubContext, getTempDir } from "./github";
 import { getContrastInputs, retrieveContrastResults } from "./contrast";
 
+interface SonarResults {
+  totalResults: number;
+  results: any
+}
+
+const MAX_PAGE_SIZE = 500;
```
Comment (carlosu7) [inline src/action.ts:20]: @drdavella if you would like to test it, you will need to update this value and upload the dist/index.js file

File: src/pixee-platform.ts [hunk 7]
```diff
@@ -5,12 +5,17 @@ import FormData from "form-data";
 import { TOOL_PATH } from "./inputs";
 import { getGitHubContext, getRepositoryInfo } from "./github";
 
-export async function uploadInputFile(tool: TOOL_PATH, file: string) {
-  const fileContent = fs.readFileSync(file, "utf-8");
+export async function uploadInputFile(tool: TOOL_PATH, files: Array<string>) {
```
Comment (drdavella) [inline src/pixee-platform.ts:8]:
```suggestion
export async function uploadInputFiles(tool: TOOL_PATH, files: Array<string>) {
```

File: __tests__/pixee-platform.test.ts [hunk 8]
```diff
@@ -38,7 +38,7 @@ describe("pixee-platform", () => {
       }
     });
 
-    await uploadInputFile("sonar_issues", file.name);
+    await uploadInputFile("sonar_issues", new Array(file.name));
```
Comment (drdavella) [inline __tests__/pixee-platform.test.ts:41]: Can we add a new test with multiple files?

**Review by drdavella — COMMENTED** [review]
@carlosu7 I think this looks good overall but I have just a few minor comments 🙏

**Review by drdavella — APPROVED** [review]

**Pull request merged**